oak_sign/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use anyhow::Context;
+use pem::Pem;
+use ring::{
+    rand,
+    signature::{self, Ed25519KeyPair, KeyPair},
+};
+use std::{
+    collections::HashMap,
+    fs::{read, write},
+};
+
+// PEM file tags.
+pub const PRIVATE_KEY_TAG: &str = "PRIVATE KEY";
+pub const PUBLIC_KEY_TAG: &str = "PUBLIC KEY";
+pub const SIGNATURE_TAG: &str = "SIGNATURE";
+pub const HASH_TAG: &str = "HASH";
+
+/// Convenience struct for private and public key.
+pub struct KeyPairStruct {
```

Comment:
Why the `Struct` suffix? Is it because `ring` already has a type with the same name? If so, I guess we could refer to that as the fully qualified name, I don't think adding `Struct` suffix on the local one is particularly meaningful (nor scalable).